providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/connector/file_watcher.py

```diff
@@ -0,0 +1,96 @@
+import json
+import logging
+import os
+import threading
+import time
+import typing
+
+import yaml
+
+from openfeature.evaluation_context import EvaluationContext
+from openfeature.event import ProviderEventDetails
+from openfeature.exception import ParseError, ProviderNotReadyError
+from openfeature.provider.provider import AbstractProvider
+
+from ..connector import FlagStateConnector
+from ..flags import FlagStore
+
+logger = logging.getLogger("openfeature.contrib")
+
+
+class FileWatcher(FlagStateConnector):
+    def __init__(
+        self,
+        file_path: str,
+        provider: AbstractProvider,
+        flag_store: FlagStore,
+        poll_interval_seconds: float = 1.0,
+    ):
+        self.file_path = file_path
+        self.provider = provider
+        self.poll_interval_seconds = poll_interval_seconds
+
+        self.last_modified = 0.0
+        self.flag_store = flag_store
+        self.emit_ready = False
```

Comment:
minor - This variable name confused me with ready event. Shall we rename this to something like `emit_on_success` ?

Reply:
I went back and forth on naming so agree. Should we be overly-explicit here?
`should_emit_ready_on_success`

Reply:
@colebaileygit I am fine with the proposal :) it's a nit but to make it easy to understand the logic